lib/ffmpeg/encoderbase.py

```diff
@@ -69,7 +69,8 @@ def gen_output_opts(self):
     if vars(self).get("minrate", None) is not None:
       opts += " -b:v {minrate}k"
     if vars(self).get("maxrate", None) is not None:
-      opts += " -maxrate {maxrate}k"
+      if "vbr_default" != self.rcmode:
+        opts += " -maxrate {maxrate}k"
```

Comment:
Please don't introduce non-standard or custom rcmode values.

Also, IIUC, if `-maxrate` is not specified then ffmpeg just sets it to target rate (e.g. bitrate)... which is actually CBR, not VBR.

Reply:
> Also, IIUC, if `-maxrate` is not specified then ffmpeg just sets it to target rate (e.g. bitrate)... which is actually CBR, not VBR.

Actually it's different behavior between ffmpeg-qsv and ffmpeg-vaapi.  So it's better to be explicit with both min and max rates.

Reply:
For maxrate, it is hard-coded as a function of bitrate in most/all tests.  You can change the test function in the "quality" suite to use the dict `setdefault` instead of `update`.  This way, if it's specified in user config then it will be used, else the hard-coded default function of bitrate will be used.

```python
vars(self).update(minrate = self.bitrate)
vars(self).setdefault("maxrate", self.bitrate * 2)
```

...then in user config, you can set `maxrate = None` in test case definition.  This is still a little awkward and a bit of a hack, but probably more obvious to anyone that reads it later on.